secretmanager/spec/snippets_spec.rb

```diff
@@ -204,6 +206,49 @@
     end
   end
 
+  describe "#iam_grant_access" do
+    it "grants access to the secret" do
+      expect(secret).to be
+      expect {
+        policy = iam_grant_access(
+          project_id: project_id,
+          secret_id:  secret_id,
+          member:     iam_user
+        )
+
+        expect(policy).to be
+        bind = policy.bindings.find do |b|
+          b.role == "roles/secretmanager.secretAccessor"
+        end
+
+        expect(bind).to be
+        expect(bind.members).to include(iam_user)
+      }.to output(/Updated IAM policy/).to_stdout
+    end
+  end
+
+  describe "#iam_revoke_access" do
```

Comment:
It appears that this depends on the test order (and possibly also that the previous test run left everything in a clean state). Can we structure the tests so they're more independent?

Reply:
`secret` is in a `let` block and it calls `create_secret`, so each test is independent.

Reply:
Here we're testing the IAM policy, however. One test assumes that `iam_user` is not a member at first, adds it, and then asserts that it is now a member. The other test assumes that `iam_user` _is_ a member at first, removes it, and then asserts that it is no longer a member.

Reply:
I guess since each secret resource is independent, then the tests are independent, but the revoke test isn't actually testing anything since `iam_user` wasn't there to begin with?